### ISSUE TITLE
Wrong duration info

### ISSUE THREAD

**Issue by Smiuv** [Zeugma440/atldotnet#282]
hello
I found another bug related to the timing of a music track. (This happened for several songs from the same album).
I will email them to you.

**Comment by Zeugma440** [comment]
Managed to repro that too. ATL gives the same wrong duration as Windows Explorer and VLC. Foobar2000 gets it right.

Gotta find what's wrong with these files...

**Comment by Smiuv** [comment]
Maybe it's because of a corruption in the file format because I found out the real time I converted it to an ogg with ffmpeg.

**Comment by Zeugma440** [comment]
I'm on it. I'm not sure it's corrupt. It looks like the MPEG frames have been encoded with loads of padding bytes.

That means the "fast" way of calculating duration (`audio data size  / bitrate * 8`) doesn't work well, and we need to parse it by reading every frame.

I'll confirm that once I get my tests running.

**Comment by Zeugma440** [comment]
Okay, the problem has absolutely nothing to do with padding.

I realized the files you sent are structured as follows :

- (File start)
- An ID3v2.4 tag
- MPEG audio data
- An ID3v2.3 tag (!)
- An ID3v1 tag
- (File end)

Problem is the ID3v2.3 tag is located at the end of the file (which is _very_ exotic, especially knowing a valid ID3v2 tag is already at its beginning) and doesn't follow the prepended tag convention, making it a "ghost" data chunk that cannot be easily detected.

The only solution I found is to read across all MPEG headers to locate the actual end of the audio data chunk, which works fine but eats up a significant overhead while reading the file.

This method is disabled by default and can be enabled by setting `Settings.MP3_parseExactDuration` to `true`.
